shm/Delegate.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <cstddef>
+#include <iostream>
+
+class Delegate {
```

Comment:
Do we need this class? Can't `setDelegate(Delegate* const delegate)` just be `setDelegate(Player* const delegate`?

Reply:
But then `Ship` class will have the access to all methods of `Player`. Do we need it? 

Also I guess that it has the educational value for Mateusz - we will learn how to inherit Interfaces, use virtual functions and so on.

Reply:
I will try something out during the weekend and get back to you

Reply:
I've created a new pull request to this branch with my implementation which is not using additional class. Check it out